openmls/src/treesync/mod.rs

```diff
@@ -210,13 +223,11 @@ impl TreeSync {
                             if let Some(private_key) = private_key.take() {
                                 own_index_option =
                                     Some(u32::try_from(node_index / 2).map_err(|_| {
-                                        TreeSyncError::from(LibraryError::custom(
-                                            "Own leaf is outside of the tree",
-                                        ))
+                                        LibraryError::custom("Own leaf is outside of the tree")
```

Comment:
If this throws an error, the leaf is not outside of the tree. It just means that the `usize` couldn't be turned into a `u32`.